include/realizations/catchment/AbstractCLibBmiAdapter.hpp

```diff
@@ -89,9 +89,32 @@ namespace models {
                     return;
                 }
                 if (!utils::FileChecker::file_is_readable(bmi_lib_file)) {
-                    this->init_exception_msg =
-                            "Can't init " + this->model_name + "; unreadable shared library file '" + bmi_lib_file + "'";
-                    throw std::runtime_error(this->init_exception_msg);
+                    //Try alternative extension...
+                    size_t idx = bmi_lib_file.rfind(".");
+                    std::string alt_bmi_lib_file;                    
+                    if(bmi_lib_file.substr(idx) == ".so"){
+                        alt_bmi_lib_file = bmi_lib_file.substr(0,idx) + ".dylib";
+                    } else if(bmi_lib_file.substr(idx) == ".dylib"){
+                        alt_bmi_lib_file = bmi_lib_file.substr(0,idx) + ".so";
+                    } else {
+                        // Try appending instead of replacing...
+                        #ifdef __APPLE__
+                        alt_bmi_lib_file = bmi_lib_file + ".dylib";
+                        #else
+                        #ifdef __GNUC__
```

Comment:
It is possible that windows compiling with Cygwin or mingw could get here. Would it be worth checking for and warning that Windows is not currently supported?

Reply:
Possibly... I have a feeling it wouldn't get that far. Maybe we should consider this: https://github.com/cygwin/cygwin-install-action ... someday.